index.js

```diff
@@ -0,0 +1,126 @@
+module.exports = WMSSource;
+
+var debug = require("debug")("wms");
+var request = require("retry-request", {
+  request: require("request")
+});
+var url = require("url");
+var tilelive = require("@mapbox/tilelive");
+
+var SphericalMercator = require("@mapbox/sphericalmercator");
+
+var retry_options = {
+  retries: 4,
+  factor: 1.7
+};
+
+var merc = new SphericalMercator({
+  size: 256
+});
+
+var SUPPORTED_FORMATS = {
+  jpg: {
+    mime: "image/jpeg"
+  },
+  png: {
+    mime: "image/png"
+  }
+};
+
+function WMSSource(uri, callback) {
+  var parsed_uri = url.parse(uri, true);
+
+  this.name = "WMSSource";
+  this.minzoom = parseInt(parsed_uri.query.minzoom || "0");
+  this.maxzoom = parseInt(parsed_uri.query.maxzoom || "14");
+  this.scale = parseInt(parsed_uri.query.scale || "1");
+  this.tilesize = parseInt(parsed_uri.query.baseTileSize || "256");
+  this.format = parsed_uri.query.format || "png";
+  this.server_root = parsed_uri.query["source"];
+
+  this.options = parsed_uri.query;
+  delete this.options.source;
+  delete this.options.minzoom;
+  delete this.options.maxzoom;
+  delete this.options.scale;
+  delete this.options.baseTileSize;
+  delete this.options.format;
+
+  callback(null, this);
+}
+
+WMSSource.prototype.getTile = function(z, x, y, callback) {
+  var tile_bbox = merc.bbox(x, y, z, false, "900913");
```

Comment:
Why not 3857?

Reply:
I guess that library is old, it wants 900913. See https://github.com/mapbox/sphericalmercator/blob/master/sphericalmercator.js#L115-L120